generator/src/test/java/com/scottlogic/deg/generator/fieldspecs/NumericRestrictionsMergeOperationTests.java

```diff
@@ -0,0 +1,165 @@
+            new NumericRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec merging = FieldSpec.Empty
+            .withTypeRestrictions(DataTypeRestrictions.createFromWhiteList(IsOfTypeConstraint.Types.STRING), FieldSpecSource.Empty);
+
+        when(merger.merge(left.getNumericRestrictions(), right.getNumericRestrictions()))
+            .thenReturn(new MergeResult<>());
+
+        Optional<FieldSpec> result = operation.applyMergeOperation(left, right, merging);
+
+        Assert.assertThat(result.isPresent(), is(true));
+        Assert.assertThat(result.get(), not(sameInstance(merging)));
+        Assert.assertThat(result.get().getNumericRestrictions(), is(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions(), not(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions().getAllowedTypes(), not(hasItem(IsOfTypeConstraint.Types.NUMERIC)));
+    }
+
+    @Test
+    public void applyMergeOperation_withContradictoryNumericRestrictionsAndNumericTypeOnlyPermittedType_shouldPreventAnyNumericValues(){
+        NumericRestrictionsMerger merger = mock(NumericRestrictionsMerger.class);
+        NumericRestrictionsMergeOperation operation = new NumericRestrictionsMergeOperation(merger);
+        FieldSpec left = FieldSpec.Empty.withNumericRestrictions(
+            new NumericRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec right = FieldSpec.Empty.withNumericRestrictions(
+            new NumericRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec merging = FieldSpec.Empty
+            .withTypeRestrictions(DataTypeRestrictions.createFromWhiteList(IsOfTypeConstraint.Types.NUMERIC), FieldSpecSource.Empty);
+
+        when(merger.merge(left.getNumericRestrictions(), right.getNumericRestrictions()))
+            .thenReturn(new MergeResult<>());
+
+        Optional<FieldSpec> result = operation.applyMergeOperation(left, right, merging);
+
+        Assert.assertThat(result.isPresent(), is(true));
+        Assert.assertThat(result.get(), not(sameInstance(merging)));
+        Assert.assertThat(result.get().getNumericRestrictions(), is(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions(), not(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions().getAllowedTypes(), empty());
+    }
+
+    @Test
+    public void applyMergeOperation_withMergableNumericRestrictions_shouldApplyMergedNumericRestrictions(){
+        NumericRestrictionsMerger merger = mock(NumericRestrictionsMerger.class);
+        NumericRestrictionsMergeOperation operation = new NumericRestrictionsMergeOperation(merger);
+        FieldSpec left = FieldSpec.Empty.withNumericRestrictions(
+            new NumericRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec right = FieldSpec.Empty.withNumericRestrictions(
+            new NumericRestrictions(),
+            FieldSpecSource.Empty);
```

Comment:
Same as above comment, I'd put the first 4 variables of each of these tests in a `beforeEach` if they are the same